### PR TITLE
Added endpoints for channel and contentnode recommendations.

### PR DESCRIPTION
**Pull request by ralphiee22**
## Summary

Added api endpoints for channel and content node recommendations.
Wrote tests for endpoints that merely return a subset of content nodes.

## TODO

- [x] Have tests been written for the new code?
- [ ] Has documentation been written/updated?
- [ ] New dependencies (if any) added to requirements file
- [ ] Add an entry to CHANGELOG.rst
- [ ] Add yourself it AUTHORS.rst if you don't appear there

## Reviewer guidance

*If you PR has a significant size, give the reviewer some helpful remarks*

## Issues addressed

List the issues solved or partly solved by the PR

## Documentation

If the PR has documentation, link the file here (either .rst in your repo or if built on Read The Docs)

## Screenshots (if appropriate)

They're nice. :)

### CODE REVIEW

File: kolibri/content/recommendations.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+from kolibri.content.models import ContentNode
+
+def recommendations_interaction_log():
+    """
+    Calculates the content that is accessed the most and returns a subset of that.
+
+    :return: Queryset of the the most frequently accessed content
+    """
+    # logs = ContentInteractionLog.objects.filter(channel_id=channel_id).all()
+    logs = ContentNode.objects.all()
```
Comment (rtibbles) [inline kolibri/content/recommendations.py]: These are not the logs you are looking for.

I have merged your logging PR to master, so you can point to the right things now!

File: kolibri/content/recommendations.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    content_count = Counter()
+    for log in logs:
+        content_count[log.content_id] += 1
```
Comment (rtibbles) [inline kolibri/content/recommendations.py]: See here for an efficient way to do this: https://docs.djangoproject.com/en/1.9/topics/db/aggregation/#cheat-sheet

Comment (rtibbles) [inline kolibri/content/recommendations.py]:
```
from django.db.models import Count
content_count_sorted = ContentInteractionLog.objects.filter(channel_id=channel_id).values('content_id').annotate(num_content=Count('content_id').order_by('-num_content')
```

I think this is what you are looking for - can do the counting and sorting in what I think is a single query.

Comment (jamalex) [inline kolibri/content/recommendations.py]: :heart:

**Comment by jamalex** [comment]
Not merging this into my refactor branch as there's still iteration needed, so just a heads up that we'll want to rebase this against master and make some tweaks once the refactor is merged.

**Comment by indirectlylit** [comment]
Ralph mentioned this is ready for review & merge

**Comment by rtibbles** [comment]
Good first pass. Can use this as the basis for recommendations on content page, and for first pass recommendations on learn page as well.

We will want to add additional filters for 'resume' and other categories of recommendation when we have logging hooked up.

**Pull request merged**